src/vm/prestub.cpp

```diff
@@ -1714,6 +1693,22 @@ PCODE MethodDesc::DoPrestub(MethodTable *pDispatchingMT)
 
     GCStress<cfg_any, EeconfigFastGcSPolicy, CoopGcModePolicy>::MaybeTrigger();
 
+#ifdef HAS_REMOTING_PRECODE
+    /**************************   REMOTING   *************************/
+    // Zapped methods which need remoting precode are saved with normal temporary entry point
+    // (so fixup or stub precode). So the first time we run into them here we need to create
+    // the remoting precode for them and patch everywhere with it.
+    if (IsZapped() && GetPrecodeType() == PRECODE_REMOTING)
```

Comment:
This means that we will always take this path - we will never take the path that does the JITing.

I think the condition should be something like:

`if (!HasStableEntryPoint() && GetPrecodeType() == PRECODE_REMOTING)`

Reply:
Right - my brain obviously didn't work well yesterday. Thanks for spotting this!